modules/nlu/src/backend/module-lifecycle/on-server-started.ts

```diff
@@ -142,11 +146,37 @@ const registerMiddleware = async (bp: typeof sdk, state: NLUState) => {
   }
 }
 
+const registerElectionMiddleware = async (bp: typeof sdk, state: NLUState) => {
+  bp.events.registerMiddleware({
+    name: 'nlu-election.incoming',
+    direction: 'incoming',
+    order: 11,
+    description: 'Perform intent election for the outputed NLU.',
+    handler: async (event: sdk.IO.IncomingEvent, next: sdk.IO.MiddlewareNextCallback) => {
+      if (ignoreEvent(bp, state, event) || !event.nlu) {
+        return next()
+      }
+
+      const { engine } = state.nluByBot[event.botId]
```

Comment:
1 - can you do a null check on engine 

2- why do you need engine for that ? Why not simply import the legacyElectionProcess function and simply call it from there ?